### PR TITLE
Remove in memory wallet

### PR DESCRIPTION
**Pull request by jamshale**
- Removes the in-memory wallet and updates tests to use an in-memory sqlite askar wallet.
- Doesn't flatten the wallet abstraction in this PR. Will do this in an additional PR.
- Runs the tests in parallel to speed up execution time.
- Updates all tests that created an in-memory (non-async) profile (wallet) to use the async askar wallet.
- test_dispatcher had a few tests that are skipped. It seems like there might be bug in this code preventing them from working.
- there was one problematic test in test_conductor that I skipped. It has a bad mock somewhere and was causing other tests to fail when run sequentially.

### CODE REVIEW

**Comment by jamshale** [comment]
Switching to a raw key dramatically sped up the tests. Whole PR Tests take less than 3 minutes now.

**Comment by jamshale** [comment]
This should be ready. It is a breaking change for external plugins that use the old in_memory_profile for testing.

**Comment by swcurran** [comment]
Wow that is a big change.  I hope you had some editor automation to help with that! LGTM, but will defer to @dbluhm @ff137 (or other maintainers) for an approval.

**Comment by dbluhm** [comment]
@jamshale if a wallet type is not specified on startup, do we spin up with an in-memory askar wallet? (if so, this should also probably be using raw key method and `Store.generate_key`)

**Comment by dbluhm** [comment]
This is great! I'm interested in taking this one step further after this PR is merged and flattening some of the wallet related abstractions now that there is just Askar.

**Review by dbluhm — APPROVED** [review]
I reviewed every non-test file in depth and skimmed the test updates. Nice work!

File: acapy_agent/admin/tests/test_admin_server.py
Comment (ff137) [inline acapy_agent/admin/tests/test_admin_server.py:119]: Seems that the rebase dropped some of the unit tests I added :-) from #3327 
I'll re-add in follow-up PR 👍

**Pull request merged**
